tests/Unit/CrudPanel/CrudPanelCreateTest.php

```diff
@@ -0,0 +1,323 @@
+    {
+        $relationFields = $this->crudPanel->getRelationFields();
+
+        $this->assertEmpty($relationFields);
+    }
+
+    public function testGetRelationFieldsWithPivot()
+    {
+        $this->crudPanel->setModel(User::class);
+        $this->crudPanel->addFields($this->userInputFieldsDotNotation, 'create');
+
+        $relationFields = $this->crudPanel->getRelationFieldsWithPivot('create');
+
+        $this->assertEquals($this->crudPanel->create_fields['roles'], array_last($relationFields));
+    }
+
+    public function testGetRelationFieldsWithPivotNoRelations()
+    {
+        $this->crudPanel->setModel(User::class);
+        $this->crudPanel->addFields($this->nonRelationshipField, 'create');
+
+        $relationFields = $this->crudPanel->getRelationFieldsWithPivot('create');
+
+        $this->assertEmpty($relationFields);
+    }
+
+    public function testSyncPivot()
+    {
+        $this->crudPanel->setModel(User::class);
+        $this->crudPanel->addFields($this->userInputFieldsManyToMany);
+        $faker = Factory::create();
+        $inputData = [
+            'name' => $faker->name,
+            'email' => $faker->safeEmail,
+            'password' => bcrypt($faker->password()),
+            'remember_token' => null,
+            'roles' => [1, 2],
+        ];
+
+        $entry = User::find(1);
+        $this->crudPanel->syncPivot($entry, $inputData);
+
+        $this->assertEquals($inputData['roles'], $entry->roles->pluck('id')->toArray());
+    }
+
+    public function testSyncPivotUnknownData()
+    {
+        $this->crudPanel->setModel(User::class);
+        $this->crudPanel->addFields($this->nonRelationshipField);
+        $faker = Factory::create();
+        $inputData = [
+            'name' => $faker->name,
+            'email' => $faker->safeEmail,
+            'password' => bcrypt($faker->password()),
+            'remember_token' => null,
+            'roles' => [1, 2],
+        ];
+
+        $entry = User::find(1);
+        $this->crudPanel->syncPivot($entry, $inputData);
+
+        $this->assertEquals(1, $entry->roles->count());
+    }
+
+    public function testSyncPivotUnknownModel()
+    {
+        $this->setExpectedException(\BadMethodCallException::class);
+
+        $this->crudPanel->setModel(User::class);
+        $this->crudPanel->addFields($this->userInputFieldsManyToMany);
+        $faker = Factory::create();
+        $inputData = [
+            'name' => $faker->name,
+            'email' => $faker->safeEmail,
+            'password' => bcrypt($faker->password()),
+            'remember_token' => null,
+            'roles' => [1, 2],
+        ];
+
+        $entry = Article::find(1);
+        $this->crudPanel->syncPivot($entry, $inputData);
+    }
+
+    private function assertEntryEquals($expected, $actual)
+    {
+        foreach ($expected as $key => $value) {
+            if (is_array($value)) {
+                $this->assertEquals(count($value), $actual->{$key}->count());
```

Comment:
Wouldn't this mean any random array of values with the same number of keys are "equal":

```
$a = [1, 2, 3];
$b = ['a', 'b', 'c'];

# Are they equal?
```